openedx/openedx-platform
# fix: timeout error for legacy page

## Description

This PR addresses the timeout errors for the legacy Videos page. This change improves performance on the page. When a user has a course with over 100 videos, the legacy and course authoring pages take a long time to load and can eventually lead to a 504 timeout error. In a previous PR the initial fetch for course videos included fetching the usage locations for a video. This data is not used in the legacy Videos page, so this PR updates to only include the data when fetching the data for the course authoring Video page. This change impacts Authors.

## Supporting information

JIRA Ticket: [TNL-11324](https://2u-internal.atlassian.net/browse/TNL-11324)
> At first, the issues involved intermittent errors from edX, initially displaying the attached error ‘*We're performing some scheduled maintenance*’, however, currently, this recurring intermittent issue persists, and they’re unable to upload videos.

## Testing instructions

1. Import a course with greater than 200 videos
2. Navigate to the legacy Videos page (may need to turn off the waffle flag for the course authoring redirect)
3. Page should load within a handful of seconds
4. Navigate to the course authoring Videos page
5. Page will load slowly.

## Deadline

None

## Other information

This only fixes the issue for the legacy page. Still looking at pagination for the Course Authoring MFE.
